test/unit/models/test/mapper/TestPreviewMapperTest.php

```diff
@@ -81,57 +82,105 @@ public function testMapEmptyTest(): void
+        );
+    }
+
+    public function testInformationalItemDetectionByCategories(): void
+    {
+        $itemId = 'itemId';
+        $sectionTitle = 'sectionTitle';
+        $categories = [
+            'x-tao-test',
+            'x-tao-itemusage-informational'
+        ];
+
+        $config = new TestPreviewConfig([TestPreviewConfig::CHECK_INFORMATIONAL => true]);
+
+        $testPart = $this->expectsTestPart('partId');
+        $section = $this->expectSection('sectionId', $sectionTitle);
+        $itemRef = $this->expectsItemRef('itemUri', $itemId, $categories);
+        $routeItem = $this->expectRouteItem($itemRef, $testPart, $section);
+
+        $this->expectsItemResource('itemUri', 'testLabel');
+
+        static::assertEquals(
+            new TestPreviewMap($this->getFullMapData($itemId, $sectionTitle, $categories, true)),
+            $this->subject->map($this->expectsTest(), $this->expectsRoute([$routeItem]), $config)
+        );
+    }
+
+    public function testInformationalItemDetectionByResponseDeclarations(): void
+    {
+        $itemId = 'itemId';
+        $sectionTitle = 'sectionTitle';
+        $categories = [
+            'x-tao-test'
+        ];
+
+        $config = new TestPreviewConfig([TestPreviewConfig::CHECK_INFORMATIONAL => true]);
+
+        $testPart = $this->expectsTestPart('partId');
+        $section = $this->expectSection('sectionId', $sectionTitle);
+        $itemRef = $this->expectsExtendedItemRef('itemUri', $itemId, $categories);
+        $routeItem = $this->expectRouteItem($itemRef, $testPart, $section);
+
+        $this->expectsItemResource('itemUri', 'testLabel');
+
+        static::assertEquals(
+            new TestPreviewMap($this->getFullMapData($itemId, $sectionTitle, $categories, true)),
+            $this->subject->map($this->expectsTest(), $this->expectsRoute([$routeItem]), $config)
+        );
+    }
+
+    private function getFullMapData(
+        string $itemId,
+        string $sectionTitle,
+        array $categories,
+        bool $isInformational = false
+    ): array {
+        $data = [
+            'scope' => 'test',
+            'parts' => [
+                'partId' => [
+                    'id' => 'partId',
+                    'label' => 'partId',
+                    'position' => 0,
+                    'isLinear' => true,
+                    'sections' => [
+                        'sectionId' => [
+                            'id' => 'sectionId',
+                            'label' => $sectionTitle,
+                            'isCatAdaptive' => false,
                             'position' => 0,
-                            'isLinear' => true,
-                            'sections' => [
-                                'sectionId' => [
-                                    'id' => 'sectionId',
-                                    'label' => $sectionTitle,
-                                    'isCatAdaptive' => false,
+                            'items' => [
```

Comment:
Diff looking weird but I just moved out the array assignment for re-using

Reply:
Yes. I wish you could extract code refactoring from your actual changes. This would be much easier to read.

Reply:
Sorry for that, unit tests carried me away ;)